.gitlab/llnl-ci-pip.yml

```diff
@@ -39,53 +41,16 @@ pip-env-setup:
     - pip install -U pip setuptools wheel
     - pip cache purge
 
-pip-build:
+.pip_build:
   stage: build
-  needs:
-    - job: pip-env-setup
-  parallel:
-    matrix:
-      - HOST : [ "quartz" ]
-        VENV_ENV_NAME: [ "quartz-env" ]
-        RUNNER_TYPE: "shell"
-      - HOST : [ "ruby" ]
-        VENV_ENV_NAME: [ "ruby-env" ]
-        RUNNER_TYPE: "shell"
-      # - HOST : [ "lassen" ]
-      #   ENVIRONMENT: [ "BATCH" ]
-      #   VENV_ENV_NAME: [ "lassen-env" ]
-      #   RUNNER_TYPE: "batch"
-  tags:
-    - $HOST
-    - $RUNNER_TYPE
   script:
     - source ${PIP_WORK_DIR}/${VENV_ENV_NAME}/bin/activate
     - pip install --no-cache-dir -r ${PIP_REQUIREMENTS}
-    - *finalize
+    - chgrp -fR exaworks ${PIP_WORK_DIR} || true
+    - chmod -fR 02770 ${PIP_WORK_DIR}    || true
```

Comment:
let's keep it within `finalize` and then should be called at the end for both `.pip_env_setup` and `.pip_build`